src/api/app/jobs/update_packages_if_dirty_job.rb

```diff
@@ -3,7 +3,7 @@ class UpdatePackagesIfDirtyJob < ApplicationJob
 
   self.priority = 10
 
-  def perform(project_id)
-    Project.find(project_id).update_packages_if_dirty
+  def perform(project)
+    project.update_packages_if_dirty
```

Comment:
This would call `.packages` on the serialized project which would probably fail also right?

Why not do what you did in `.refresh_dirty` here.

```
def perform(project_id)
  project = Project.find(project_id)
  project.update_packages_if_dirty if project
end
```